pkg/dnet/kpfconn.go

```diff
@@ -27,6 +27,7 @@ import (
 
 type k8sPortForwardDialer struct {
 	// static
+	logCtx          context.Context
```

Comment:
I'm not sure adding this is right.  9 times out of 10, storing a context in a struct is the wrong thing.  Since we're only using it for logging, it doesn't matter if it's canceled; and so storing a persistent logCtx only serves to obscure where `Dial` is being called from (not that we call it from very many places).

Reply:
I don't love it either, but I couldn't find any other way to make a logger configured context available to the dialer. The context passed in `DialContext()` is NOT propagated to the dialer. So I added this and specifically named it `logCtx` to indicate that this context is indeed only intended for logging. Do you have a better proposal?

Reply:
Wait, really?

Reply:
To clarify, this change has nothing to do with cancelled contexts. It's needed because the context that the dialer gets, is not configured for logging. It's created here: https://github.com/datawire/grpc-go/blob/master/clientconn.go#L147

Reply:
I guess that `context.Background()` is a perfect candidate for a `dcontext.WithoutCancel(ctx)`, but perhaps a depencendy to `dlib` is undesired here?

Reply:
Wow, really.  Huh. OK.  I guess that's another PR that I should file against the gRPC library :/